### PR TITLE
release 1.3.0

### PR DESCRIPTION
**Pull request by Borda**
## What does this PR do?

version++ and other cleaning

## Before submitting
- [x] Was this discussed/approved via a GitHub issue? (not for typos and docs)
- [x] Did you read the [contributor guideline](https://github.com/PyTorchLightning/pytorch-lightning/blob/master/.github/CONTRIBUTING.md), **Pull Request** section?
- [ ] Did you make sure your PR does only one thing, instead of bundling different changes together?
- [ ] Did you make sure to update the documentation with your changes? (if necessary)
- [ ] Did you write any new necessary tests? (not for typos and docs)
- [ ] Did you verify new and existing tests pass locally with your changes?
- [x] Did you update the [CHANGELOG](https://github.com/PyTorchLightning/pytorch-lightning/blob/master/CHANGELOG.md)? (not for typos, docs, test updates, or internal minor changes/refactorings)

## PR review
Anyone in the community is free to review the PR once the tests have passed.
Before you start reviewing make sure you have read [Review guidelines](https://github.com/PyTorchLightning/pytorch-lightning/wiki/Review-guidelines). In short, see the following bullet-list:

 - [x] Is this pull request ready for review? (if not, please submit in draft mode)
 - [x] Check that all items from **Before submitting** are resolved
 - [x] Make sure the title is self-explanatory and the description concisely explains the PR
 - [x] Add labels and milestones (and optionally projects) to the PR so it can be classified

## Did you have fun?
Make sure you had fun coding 🙃

### CODE REVIEW

File: .github/workflows/ci_test-conda.yml [hunk 1]
```diff
@@ -16,7 +16,7 @@ jobs:
       matrix:
         # os: [ubuntu-20.04]
         python-version: [3.7]
-        pytorch-version: [1.4, 1.5, 1.6, 1.7, 1.8, 1.9]
+        pytorch-version: [1.4, 1.5, 1.6, 1.7, 1.8]  # todo: add back 1.9
```
Comment (Borda) [inline .github/workflows/ci_test-conda.yml:19]: this is just to have green badges or #7396 is fixed

Comment (awaelchli) [inline .github/workflows/ci_test-conda.yml:19]: I'll let you know about it 
#7396

**Review by awaelchli — APPROVED** [review]

**Review by carmocca — APPROVED** [review]
🎉

**Review by tchaton — APPROVED** [review]
LGTM !

**Review by kaushikb11 — APPROVED** [review]
💯

**Comment by leezu** [comment]
@Borda let's get https://github.com/PyTorchLightning/pytorch-lightning/pull/7362 merged first?

**Pull request merged**